qemu/qmp_test.go

```diff
@@ -1794,3 +1794,22 @@ func TestExecQomGet(t *testing.T) {
 	q.Shutdown()
 	<-disconnectedCh
 }
+
+// Checks dump-guest-memory
+func TestExecuteDumpGuestMemory(t *testing.T) {
+	connectedCh := make(chan *QMPVersion)
+	disconnectedCh := make(chan struct{})
+	buf := newQMPTestCommandBuffer(t)
+	buf.AddCommand("dump-guest-memory", nil, "return", nil)
+	cfg := QMPConfig{Logger: qmpTestLogger{}}
+	q := startQMPLoop(buf, cfg, connectedCh, disconnectedCh)
+	checkVersion(t, connectedCh)
+
+	err := q.ExecuteDumpGuestMemory(context.Background(), "file:/tmp/dump.xxx.yyy", false, "elf")
```

Comment:
- Please can you use a temporary file rather than a fixed one (`tmpfile, err := ioutil.TempFile("", "")` for example).
- It might be worth checking the file actually gets created and has size > 0?
- Can you add a `defer` to remove this (potentially huge) file at the end of the test.

Reply:
Hi @jodh-intel , `qmpTestCommandBuffer` is used to simulate QMP server, and in fact there is no QEMU processing running for the tests, nor any file will be really created.

Reply:
Ah - that makes more sense. In which case, can  you assert that `file:/tmp/dump.xxx.yyy` is *not* created?

Reply:
The file will not be created, it's only a parameter for the function signature, and there is no dump operation. So I think we don't need the check.